### PR TITLE
Add storage size and raw size to ORC stats builders

### PR DESCRIPTION
**Pull request by sdruzkin**
## Description
This change adds size and raw size to ORC column statistic builders. The new 
values and methods are not used anywhere yet.

I will be making this change in the following order to reduce the scope of changes per PR:
[x] Add size and row size to stats builders
[] Make column writers write raw sizes
[] Propagate size and raw size to the ColumnStat objects created by stats builders
[] Update DWRF protobuf if needed
[] Make DWRF metadata writer / reader translate stats from DWRF to ColumnStat
[] Populate size and rawSize in column stats during the Metastore commit 

## Motivation and Context
Close this feature gap between Velox/BBIO and Presto written files. It has been
requested multiple times for data quality and column size monitoring purposes.
See T161955910 for more context. 

## Impact
Adding size and raw size to column stats will slightly increase memory consumption
by 16 bytes per column per written stripe.

## Test Plan
Existing tests. Full test coverage will be added step 3 when new columns will be added to
the column stats and builders start populating this value.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/BinaryStatisticsBuilder.java [hunk 1]
```diff
@@ -25,6 +25,8 @@ public class BinaryStatisticsBuilder
         implements SliceColumnStatisticsBuilder
 {
     private long nonNullValueCount;
+    private long size;
```
Comment (sdruzkin) [inline presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/BinaryStatisticsBuilder.java:28]: If I remember correctly we don't use a base class for stats builders to save some memory used by inheritance.

**Review by NikhilCollooru — APPROVED** [review]
LGTM

**Comment by sdruzkin** [comment]
I decided to replace Math.addExact with regular += to allow overflow and to keep it nice and fast. Reason is that if someone succeeded in writing 9 exabytes of data, the query should not fail because of some optional column stats.

**Comment by NikhilCollooru** [comment]
> I decided to replace Math.addExact with regular increment to allow overflow and to keep it nice and fast. Reason is that if someone succeeded in writing 9 exabytes of data, the query should not fail because of some optional column stats.

Basically whether to throw exception and fail query when overflow happens or let the query succeed but have incorrect column stats.

**Comment by sdruzkin** [comment]
presto-orc tests succeeded, but 4 other tests, mostly recently added by @ajaygeorge are failing.

**Pull request merged**